cpp/src/fil/fil.cu

```diff
@@ -886,44 +960,53 @@ void tl2fil_sparse(std::vector<int>* ptrees, std::vector<fil_node_t>* pnodes,
   }
   size_t total_nodes = ptrees->back() + model.trees.back().num_nodes;
 
+  if (params->leaf_algo == VECTOR_LEAF) {
+    vector_leaf->resize(total_nodes * params->num_classes);
```

Comment:
the latter requires to add a scan during import to know which leaf gets which slot. It doesn't sound like a huge deal to me, but definitely not a single-line change. @RAMitchell ?

Reply:
This can be done similar to how it's already done with `p_trees`, but adding the number of leaves (`(num_tree_nodes + 1) / 2`) instead of the number of nodes each time.

Reply:
It's a bit of a pain to do it in a sparse way as there is some multithreaded code. I'd leave it for now.